lib/core/config/network_config.dart

```diff
@@ -1,32 +1,35 @@
 part of '../core.dart';
 
 /// Manage all url creation and building
-// ignore_for_file: avoid_classes_with_only_static_members
-class NetworkConfig {
-  // base static url
-  static final _baseStaticUrl = dotenv.get('BASE_STATIC_ENDPOINT_URL');
+// base static url
+final _baseStaticUrl = dotenv.get('BASE_STATIC_ENDPOINT_URL');
 
+extension UserProfileUrlX on UserProfile? {
```

Comment:
What about just having this on the UserProfile instead of having them as extensions? You're using `dotenv.get` anyhow.

Reply:
Sure

Reply:
Seems it won't play nicely with nulls. Unless i still keep it as an extension

Reply:
What did you try?